storage/provider/dirfuncs.go

```diff
@@ -79,14 +82,91 @@ func (o *osDirFuncs) bindMount(source, target string) error {
 	return err
 }
 
+const mountInfoPath = "/proc/self/mountinfo"
+
+func (o *osDirFuncs) infoReader() (io.ReadSeeker, func(), error) {
+	if o.mountInfoRdr == nil {
```

Comment:
why keep the mount info object open
It seems very weird that your 'closer' return is just a seek rather than an actual close.

AFAICT you actually don't have *any* logic anywhere that actually calls mountInfoRdr.Close() [say on teardown of something].
That means we actually will leak file handles in potentially all of these places:
```
$ rg "osDirFuncs\{"
loop.go
62:             &osDirFuncs{lp.run},

managedfs.go
53:             &osDirFuncs{logAndExec},

export_test.go
25:             osDirFuncs{run},
45:             osDirFuncs{run},
116:            osDirFuncs{run},
130:                    osDirFuncs{run},

tmpfs.go
66:             &osDirFuncs{p.run},

rootfs.go
64:             &osDirFuncs{p.run},
```